smarts/core/controllers/imitation_controller.py

```diff
@@ -35,15 +35,15 @@ def perform_action(
         cls,
         dt: float,
         vehicle,
-        action: Union[float, Tuple[Tuple[float, float], float]],
+        action: Union[float, Tuple[float, float]],
     ):
         """Performs an action adapting to the underlying chassis.
         Args:
             dt (float):
                 A delta time value.
             vehicle (Vehicle):
                 The vehicle to control.
-            action (Union[float, Tuple[Tuple[float, float], float]]):
+            action (Union[float, Tuple[float, float]]):
```

Comment:
This doesn't look right.  I think this should be `Union[float, Tuple[float, float]]`.
You can either set the initial speed with the singleton `float`, or you can pass in a tuple with `(acceleration, angular_velocity)`.

Reply:
That is what I did, I realised that I had originally gave it the wrong typehint and corrected it to `Union[float, Tuple[float, float]]`.